src/graphql/types/object/btc-wallet.ts

```diff
@@ -48,13 +50,35 @@ const BtcWallet = GT.Object<Wallet>({
         const { result: transactions, error } = await Wallets.getTransactionsForWallets([
           source,
         ])
-        if (error instanceof Error || transactions === null) {
-          throw error
-        }
+        if (error instanceof Error) throw mapError(error)
+        if (transactions === null) throw error
```

Comment:
this should remains `if (error instanceof Error || transactions === null)`. We should not return an error if there are transactions

Reply:
This was the original behaviour, I just decomposed it a bit to be able to map `error`.

Should we throw another error explicitly instead? Any ideas on this @samerbuna?

Reply:
Ok so `transactions` can only be null if `error` is an instanceof Error, but it's tricky to get the type checker to understand this relationship so this conditional should never actually throw, but is just to signal to the type checker that `transactions` passed into the function just below is of type `WalletTransaction[]`

Reply:
Added a comment in case we forget again: https://github.com/GaloyMoney/galoy/pull/1511/commits/8ccde29578a77d7d469c0344f6337c82fdf11083